src/js/messaging/containers/EmailNotifications.jsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import classNames from 'classnames';
+
+import ErrorableRadioButtons from '../../common/components/form-elements/ErrorableRadioButtons';
+import ErrorableTextInput from '../../common/components/form-elements/ErrorableTextInput';
+import LoadingIndicator from '../../common/components/LoadingIndicator';
+import { makeField } from '../../common/model/fields';
+
+import {
+  fetchPreferences,
+  savePreferences,
+  setNotificationEmail,
+  setNotificationFrequency
+} from '../actions';
+
+import ModalDiscardChanges from '../components/ModalDiscardChanges';
+
+export class EmailNotifications extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.renderSaveButtons = this.renderSaveButtons.bind(this);
+
+    this.state = { discardChanges: false };
```

Comment:
Most of the modal states are in the Redux store, but wanted to keep this in the component's state. My thinking was that if the modal's usage is limited to a particular parent component, we don't need to track it in the more global Redux state. This limits the number of actions (open and close for each and every modal; yuck) we need to manage as well.